src/Symfony/Bridge/Twig/AppVariable.php

```diff
@@ -145,4 +145,35 @@ public function getDebug()
 
         return $this->debug;
     }
+
+    /**
+     * Returns some or all the existing flash messages:
+     *  * getFlashes() returns all the flash messages
+     *  * getFlashes('notice') returns a simple array with flash messages of that type
+     *  * getFlashes(array('notice', 'error')) returns a nested array of type => messages
+     *
+     * @return array
+     */
+    public function getFlashes($types = null)
+    {
+        // needed to avoid starting the session automatically when looking for flash messages
+        try {
+            $session = $this->getSession();
+            if (null !== $session && !$session->isStarted()) {
+                return array();
+            }
+        } catch (\RuntimeException $e) {
+            return array();
+        }
+
+        if (null === $types) {
+            return $session->getFlashBag()->all();
+        }
+
+        if (1 === count($types)) {
+            return $session->getFlashBag()->get($types[0]);
+        }
```

Comment:
Oh and it violates the api (more or less) with `getFlashes(['notice'])` (i'd expect a nested array... but you get a flat one).